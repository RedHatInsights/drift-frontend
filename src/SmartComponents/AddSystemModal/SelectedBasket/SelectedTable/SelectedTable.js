import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import SelectedBasketCheckbox from '../SelectedBasketCheckbox/SelectedBasketCheckbox';

export class SelectedTable extends Component {
    constructor(props) {
        super(props);
    }

    buildRows = () => {
        const { entities, findType, selectedBaselineContent, selectedHSPContent, selectedSystemContent, handleDeselect } = this.props;
        let rows = [];

        selectedBaselineContent.forEach(function(baseline) {
            rows.push(
                <tr>
                    <td>
                        <SelectedBasketCheckbox
                            type='baseline'
                            findType={ findType }
                            id={ baseline.id }
                            handleDeselect={ handleDeselect }
                        />
                    </td>
                    <td>{ baseline.icon }</td>
                    <td style={{ 'align-content': 'left' }}>{ baseline.name }</td>
                </tr>
            );
        });

        selectedSystemContent.forEach(function(system) {
            rows.push(
                <tr>
                    <td>
                        <SelectedBasketCheckbox
                            type='system'
                            findType={ findType }
                            id={ system.id }
                            handleDeselect={ handleDeselect }
                        />
                    </td>
                    <td>{ system.icon }</td>
                    <td style={{ 'align-content': 'left' }}>{ system.name }</td>
                </tr>
            );

            selectedHSPContent?.forEach(function(hsp) {
                if (hsp.system_id === system.id) {
                    rows.push(
                        <tr>
                            <td>
                                <SelectedBasketCheckbox
                                    type='hsp'
                                    findType={ findType }
                                    id={ hsp.id }
                                    handleDeselect={ handleDeselect }
                                />
                            </td>
                            <td>{ hsp.icon }</td>
                            <td style={{ 'align-content': 'left' }}>
                                <div>{ system.name }</div>
                                <div>{ moment.utc(hsp.captured_date).format('DD MMM YYYY, HH:mm UTC') }</div>
                            </td>
                        </tr>
                    );
                }
            });
        });

        selectedHSPContent?.forEach(function(hsp) {
            if (!entities?.selectedSystemIds?.includes(hsp.system_id)) {

                rows.push(
                    <tr>
                        <td>
                            <SelectedBasketCheckbox
                                type='hsp'
                                findType={ findType }
                                id={ hsp.id }
                                handleDeselect={ handleDeselect }
                            />
                        </td>
                        <td>{ hsp.icon }</td>
                        <td style={{ 'align-content': 'left' }}>
                            <div>{ hsp.system_name }</div>
                            <div>{ moment.utc(hsp.captured_date).format('DD MMM YYYY, HH:mm UTC') }</div>
                        </td>
                    </tr>
                );
            }
        });

        return rows;
    }

    render() {
        return (
            <React.Fragment>
                <table className='pf-c-table'>
                    <tbody>
                        { this.buildRows() }
                    </tbody>
                </table>
            </React.Fragment>
        );
    }
}

SelectedTable.propTypes = {
    selectedBaselineContent: PropTypes.array,
    entities: PropTypes.object,
    selectedHSPContent: PropTypes.array,
    selectedSystemContent: PropTypes.array,
    findType: PropTypes.func,
    handleDeselect: PropTypes.func
};

export default SelectedTable;
