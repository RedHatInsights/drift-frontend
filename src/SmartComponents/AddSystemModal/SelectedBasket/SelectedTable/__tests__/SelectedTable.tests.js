/*eslint-disable camelcase*/
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BlueprintIcon, ClockIcon, ServerIcon } from '@patternfly/react-icons';

import SelectedTable from '../SelectedTable';

describe('SelectedTable', () => {
    let props;

    beforeEach(() => {
        props = {
            selectedBaselineContent: [
                { id: 'abcd1234', name: 'baseline', icon: <BlueprintIcon /> }
            ],
            selectedHSPContent: [
                {
                    id: 'ijkl9101',
                    captured_date: '2021-03-03T06:40:32+00:00',
                    icon: <ClockIcon />,
                    system_id: 'efgh5678',
                    system_name: 'system'
                }
            ],
            selectedSystemContent: [
                { id: 'efgh5678', name: 'system', icon: <ServerIcon /> }
            ],
            entities: {
                rows: [
                    { id: 'efgh5678' }
                ]
            },
            findType: jest.fn(),
            handleDeselect: jest.fn()
        };
    });

    it('should render empty', () => {
        props.selectedSystemContent = [];
        props.selectedBaselineContent = [];
        props.selectedHSPContent = [];

        const wrapper = shallow(
            <SelectedTable
                { ...props }
            />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with system, baseline and hsp', () => {
        const wrapper = shallow(
            <SelectedTable
                { ...props }
            />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render hsp with system and hsp without system', () => {
        props.selectedHSPContent.push({
            id: 'mnop1121',
            captured_date: '2021-03-02T06:40:32+00:00',
            icon: <ClockIcon />,
            system_id: '4321dcba',
            system_name: 'system2'
        });

        props.entities.rows.push({ id: '4321dcba' });
        props.entities.selectedSystemIds = [ 'efgh5678' ];

        const wrapper = shallow(
            <SelectedTable
                { ...props }
            />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
/*eslint-enable camelcase*/
