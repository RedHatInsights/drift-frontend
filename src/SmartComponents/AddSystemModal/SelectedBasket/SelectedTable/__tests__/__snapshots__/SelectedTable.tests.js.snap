// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedTable should render empty 1`] = `
<Fragment>
  <table
    className="pf-c-table"
  >
    <tbody />
  </table>
</Fragment>
`;

exports[`SelectedTable should render hsp with system and hsp without system 1`] = `
<Fragment>
  <table
    className="pf-c-table"
  >
    <tbody>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="abcd1234"
            type="baseline"
          />
        </td>
        <td>
          <BlueprintIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          baseline
        </td>
      </tr>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="efgh5678"
            type="system"
          />
        </td>
        <td>
          <ServerIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          system
        </td>
      </tr>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="ijkl9101"
            type="hsp"
          />
        </td>
        <td>
          <ClockIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          <div>
            system
          </div>
          <div>
            03 Mar 2021, 06:40 UTC
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="mnop1121"
            type="hsp"
          />
        </td>
        <td>
          <ClockIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          <div>
            system2
          </div>
          <div>
            02 Mar 2021, 06:40 UTC
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;

exports[`SelectedTable should render with system, baseline and hsp 1`] = `
<Fragment>
  <table
    className="pf-c-table"
  >
    <tbody>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="abcd1234"
            type="baseline"
          />
        </td>
        <td>
          <BlueprintIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          baseline
        </td>
      </tr>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="efgh5678"
            type="system"
          />
        </td>
        <td>
          <ServerIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          system
        </td>
      </tr>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="ijkl9101"
            type="hsp"
          />
        </td>
        <td>
          <ClockIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          <div>
            system
          </div>
          <div>
            03 Mar 2021, 06:40 UTC
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <SelectedBasketCheckbox
            findType={[MockFunction]}
            handleDeselect={[MockFunction]}
            id="ijkl9101"
            type="hsp"
          />
        </td>
        <td>
          <ClockIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </td>
        <td
          style={
            Object {
              "align-content": "left",
            }
          }
        >
          <div>
            system
          </div>
          <div>
            03 Mar 2021, 06:40 UTC
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
